### ISSUE TITLE
Improve anatomy JSON Scheme and Display

### ISSUE THREAD

**Issue by lukewarlow** [openui/open-ui#845]
Currently the anatomy that can be defined for a component in a design system is rather limited. Just an array of names with a description.

It's also displayed rather simply

![image](https://github.com/openui/open-ui/assets/32498324/ce4977b5-e789-4c91-8c7c-a6a0b7f6ffcd)

It would be good to be able to represent the structure in an anatomy like we can in our anatomy proposals. It would also be good to display the anatomies in a way more akin to the concepts where the various design systems have their anatomies displayed and labelled with where the anatomy comes from.

This would make it far easier to turn these anatomies into a proposal with confidence that we cover X% of design systems.

**Comment by lukewarlow** [comment]
A syntax such as that below could be used instead allowing you to show nesting and then this could be converted to an actual HTML representation of the anatomy for display in the UI. Would have to think pseudo and legacy-pseudo would be optional and only really useful for browsers. Perhaps a more generic name could be thought of that would be shared with other design systems "slots and parts". But just an initial idea?


```json
      "anatomy": [
        {
          "name": "slider-container",
          "legacy-pseudo": "-webkit-slider-container",
          "children": [
            {
              "name": "track",
              "pseudo": "slider-track",
              "legacy-pseudo": "-webkit-slider-runnable-track",
              "children": [
                {
                  "name": "thumb",
                  "pseudo": "slider-thumb",
                  "legacy-pseudo": "-webkit-slider-thumb"
                }
              ]
            }
          ]
        }
      ],
```

**Comment by gregwhitworth** [comment]
@lukewarlow as we you were presenting this it dawned on me that we may want to actually standardize a web component schema. Then a colleague internally who works on one of our builders reached out about the potential for something along these lines and I remembered the WC manifest spec. Should we align schema with that proposal: https://github.com/webcomponents/custom-elements-manifest

**Comment by lukewarlow** [comment]
Huh that looks pretty neat. I think that's a good idea. Will be quite exhaustive that way.

**Comment by gregwhitworth** [comment]
@lukewarlow I don't think we need to discuss this at this moment; however I'm intrigued at your "quite exhaustive" statement as maybe we should influence that spec/proposal for changes? For now just land in the direction you were already heading and possibly do a draft that follows that proposal? @mfreed7 fyi since you requested the naming of "parts and slots" issue and if we do decide to adopt the WCCG schema then it may already be covered.

**Comment by lukewarlow** [comment]
I noticed our schemas were lacking in other places https://github.com/openui/open-ui/issues/875  that I think that above one could handle (events for example) and others that it might not handle (custom state?) that we can upstream.

In that case I have the open PR that we can land get the browser follow up one in aswell and then I can work on a draft using the linked schema.

I'm happy to iterate on this and migrate between them as time goes on so the duplicate work isn't an issue for me.

**Comment by lukewarlow** [comment]
@gregwhitworth on the topic of upstreaming improvements, I've made a [PR](https://github.com/webcomponents/custom-elements-manifest/pull/128) to add custom state to the schema which we can use for both design systems proper and also for browser pseudo classes.

**Comment by lukewarlow** [comment]
THe PR mentioned above got merged. So when I've got some time I'll pick this back up and see what I can come up with.
